src/object/schema_manager.c

```diff
@@ -3312,10 +3321,22 @@ sm_check_system_class_by_name (const char *name)
       return false;
     }
 
-  count = sizeof (system_classes) / sizeof (system_classes[0]);
-  for (i = 0; i < count; i++)
+  /* binary search */
+  left = 0;
+  right = count;
+  while (left <= right)
     {
-      if (strncmp (downcase_name, system_classes[i].name, system_classes[i].len) == 0)
```

Comment:
how about below?
`if (memcmp (downcase_name, system_classes[i].name, system_classes[i].len+1) == 0)`

Reply:
If it was the old code, it would have been a good way.

I improved it by using the length comparison method a while ago.
So it seems there is no problem to use strcmp ().